### DISCUSSION TITLE
Creating Modules Via Code instead of CLI

### DISCUSSION THREAD

**Discussion by ali-amento** [nWidart/laravel-modules#2000] [Q&A]
### Description
I am working on a feature for my app that allows module installation through the admin panel using Livewire. The process includes uploading a ZIP file, creating a plain module, unzipping it, and running migrations and seeders. However, I am facing issues when executing the Artisan::call commands in my Livewire component.

### Steps I am following:

1. When a new ZIP file is uploaded, I create a new plain module using the module:make Artisan command:
       `Artisan::call('module:make', ['name' => [$module], '--plain' => true]);`
2. After that, I unzip the contents of the module into the appropriate folder, overriding the existing code.
3. Then, I attempt to run migrations and seeders using these commands:
     `Artisan::call('module:migrate', ['module' => $addonInfo->name]);
       Artisan::call('module:seed', ['module' => $addonInfo->name]);`

### Problem:

- The Artisan::call commands work fine in the command-line interface (CLI), but when executed via Livewire in the admin panel, they don't seem to function as expected.
- Migration is not being called and I am getting the error: Module not found.
- After reloading the page, the entire site throws the error: Class "Modules\Forumwise\Providers\ForumwiseServiceProvider" not found.

### Questions:

- Why is the migration not being called, and why is the module not recognized?
- What might be causing the Class "Modules\Forumwise\Providers\ForumwiseServiceProvider" not found error?
- Is there something I’m missing to make these Artisan commands work smoothly from Livewire?

**Any help or suggestions on how to smooth this process would be greatly appreciated!**

**Comment by dcblogdev** [comment]
you may need an additional step to "update" the cache by calling "module:clear-compiled" or "composer dump-autoload" before trying to use the module, also I think the module has to be enabled to migrate, so its worth checking its enabled.

**Reply by ali-amento** [reply]
These issues seem to occur as secondary effects, but the main concern is with the 
`Artisan::call('module:make', ['name' => [$module], '--plain' => true]);`
command. In the CLI, this command takes time to execute, but when called from the controller, it runs instantly. I suspect the module isn't being created properly, which is why it's not working right away. I tried manually calling:

`Module::register();`
`Module::boot();`

But it didn’t resolve the issue. However, there seems to be a delay or some kind of "magic" where, after a short time or upon page refresh, it starts working as expected.
